utils_cv/tracking/model.py

```diff
@@ -163,18 +120,23 @@ def __init__(
         self.opt.device = torch_device()
 
         self.dataset = dataset
-        self.model = self.init_model()
         self.model_path = model_path
+        self.model = self.init_model()
 
     def init_model(self) -> nn.Module:
         """
         Download and initialize the baseline FairMOT model.
         """
-        model_dir = osp.join(self.opt.root_dir, "models")
-        baseline_path = osp.join(model_dir, "all_dla34.pth")
-        #         os.makedirs(model_dir, exist_ok=True)
-        #         _download_baseline(BASELINE_URL, baseline_path)
-        self.opt.load_model = baseline_path
+        if osp.isfile(self.model_path):
+            self.opt.load_model = self.model_path
+        else:
+            baseline_path = osp.join(
```

Comment:
This is a bit funny. If "model_path" is a file then use it, otherwise it gets totally ignored. Maybe instead check if "model_path" is None?

Reply:
The thinking behind this was that a user may want to initialize a TrackingLearner only to use for inference. In this case they could supply `model_path` (an existing file) as a param. Otherwise, we default to `all_dla34.pth` but still ask for `model_path` to be specified so that we can save the finetuned weights to that path when `fit` is called. (Recall that `predict` currently depends on the src `JDETracker`, which must read the model weights from disk). 

I agree it is a bit weird, am open to suggestions.

Reply:
OK for now, but def weird to over-write a passed in model (if I understand correctly). How about as discussed we add the functionality to load a model in JDETracker by allowing a model object to be passed in (I realize this means we slightly edit the original FairMOT code). That way we don't need to save things anymore in fit, and also we can use/wrap the existing model loading functionality in FairMOT